### PR TITLE
Update git configs

### PR DESCRIPTION
**Pull request by cheenamalhotra**
We've been getting intermittent errors when performing `git fetch distro` in pipelines on multiple OSes, stages/jobs.
The error messages are as:
- error: RPC failed; HTTP 500 curl 56 GnuTLS recv error (-54): Error in the pull function. ([link](https://mssqltools.visualstudio.com/CrossPlatBuildScripts/_build/results?buildId=208194&view=logs&j=bd1e231b-cef9-51a2-e461-b67194c0a817&t=ab758b51-335c-5fef-2682-4f91bf9267f0&l=12))
- error: RPC failed; HTTP 500 curl 56 Recv failure: Connection reset by peer 
fatal: expected 'acknowledgments' ([link](https://mssqltools.visualstudio.com/CrossPlatBuildScripts/_build/results?buildId=208202&view=logs&j=3792f238-f35e-5f82-0dbc-272432d9a0fb&t=77130470-8bc4-5ef7-e47e-b9c1547cf69d&l=12))
- error: RPC failed; HTTP 500 curl 56 Recv failure: Connection reset by peer
fatal: expected 'packfile' ([link](https://mssqltools.visualstudio.com/CrossPlatBuildScripts/_build/results?buildId=208202&view=logs&j=7c206f6d-993c-583c-875a-239a70ef97fe&t=d2cc5050-2748-5724-49ee-fab7041bf932&l=13))
- error: RPC failed; HTTP 500 curl 56 GnuTLS recv error (-54): Error in the pull function.
fatal: error reading section header 'acknowledgments' ([link](https://mssqltools.visualstudio.com/CrossPlatBuildScripts/_build/results?buildId=208202&view=logs&j=4d2898ab-dfbe-557e-92e7-aaac158fdd2f&t=d4147bd7-cefa-5b93-8666-730837d89c85&l=13))
- error: RPC failed; HTTP 500 curl 56 Recv failure: Connection was reset
fatal: expected 'acknowledgments'
merge: 72f281cdcc5d92e168d3ce089814978e9701a67e - not something we can merge ([link](https://mssqltools.visualstudio.com/CrossPlatBuildScripts/_build/results?buildId=208202&view=logs&j=471dc74f-2452-533b-c058-e43cd1b98abf&t=bb53992a-5086-53f3-b974-02c2255cec6a&l=14))

The errors are not consistent and point to incomplete download over network or network interruption due to unknown reason. Increasing `http(s).postbuffer` seems to help. 

[Passing build 1](https://mssqltools.visualstudio.com/CrossPlatBuildScripts/_build/results?buildId=208225&view=logs&j=bd1e231b-cef9-51a2-e461-b67194c0a817&t=ab758b51-335c-5fef-2682-4f91bf9267f0)
[Passing build 2](https://mssqltools.visualstudio.com/CrossPlatBuildScripts/_build/results?buildId=208229&view=logs&j=bd1e231b-cef9-51a2-e461-b67194c0a817&t=ab758b51-335c-5fef-2682-4f91bf9267f0)

### CODE REVIEW

File: build/azure-pipelines/darwin/sql-product-build-darwin.yml [hunk 1]
```diff
@@ -41,12 +41,14 @@ steps:
 
       git config user.email "sqltools@service.microsoft.com"
       git config user.name "AzureDataStudio"
+      git config --global http.postBuffer 524288000
```
Comment (Charles-Gagnon) [inline build/azure-pipelines/darwin/sql-product-build-darwin.yml:44]: Please add comments linking to this issue for context

**Review by Charles-Gagnon — APPROVED** [review]
No idea why this just started happening now though?

**Comment by alanrenmsft** [comment]
@cheenamalhotra I also did the same thing yesterday, any idea why was it not working for me?

**Comment by alanrenmsft** [comment]
feels like it was network issue with ADO agents, it started passing even without your changes.
image

**Comment by cheenamalhotra** [comment]
It’s possible. But this configuration is harmless and good for faster download, so we can let it stay!

**Pull request merged**